code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,266 @@
+
+/// Humans caught in the aoe_leash will be pulled back if they leave it's radius
+/obj/structure/xeno/aoe_leash/proc/check_dist(datum/leash_victims, atom/oldloc)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/victim = leash_victims
+	var/distance = get_dist(victim, loc)
+	if(distance >= leash_radius)
+		victim.Move(oldloc)
+
+/obj/structure/xeno/aoe_leash/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(X.status_flags & INCORPOREAL)
+		return
+	X.visible_message(span_xenonotice("\The [X] starts tearing down \the [src]!"), \
+	span_xenonotice("We start to tear down \the [src]."))
+	if(!do_after(X, 2 SECONDS, TRUE, X, BUSY_ICON_GENERIC))
+		return
+	if(!istype(src)) // Prevent jumping to other turfs if do_after completes with the wall already gone
+		return
+	X.do_attack_animation(src, ATTACK_EFFECT_CLAW)
+	X.visible_message(span_xenonotice("\The [X] tears down \the [src]!"), \
+	span_xenonotice("We tear down \the [src]."))
+	playsound(src, "alien_resin_break", 25)
+	take_damage(max_integrity)
+
+// ***************************************
+// *********** Spiderling Section
+// ***************************************
+
+/datum/action/xeno_action/create_spiderling
+	name = "Birth Spiderling"
+	ability_name = "birth_spiderling"
+	mechanics_text = " Spawn a spiderling directly under you"
+	action_icon_state = "spawn_hugger" // temporary until I get my own icons
+	plasma_cost = 1 // increase later
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_LEASH_BALL
+	var/list/mob/living/carbon/xenomorph/spiderling/spiderlings
+
+/datum/action/xeno_action/create_spiderling/New(Target)
+	. = ..()
+	spiderlings = list()
+
+/datum/action/xeno_action/create_spiderling/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!do_after(X, 0.5 SECONDS, TRUE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+
+	var/new_spiderling = new /mob/living/carbon/xenomorph/spiderling(owner.loc, owner, owner)
+	add_spiderling(new_spiderling)
+	succeed_activate()
+	add_cooldown()
+
+/datum/action/xeno_action/create_spiderling/proc/add_spiderling(mob/living/carbon/xenomorph/spiderling/new_spiderling)
+	RegisterSignal(new_spiderling, COMSIG_MOB_DEATH, .proc/remove_spiderling)
+	spiderlings += new_spiderling
+
+/datum/action/xeno_action/create_spiderling/proc/remove_spiderling(datum/source)
+	SIGNAL_HANDLER
+	spiderlings -= source
+	var/datum/action/xeno_action/spider_swarm/spider_swarm_action = owner.actions_by_path[/datum/action/xeno_action/spider_swarm]
+	if(!spider_swarm_action)
+		return
+	if(source != spider_swarm_action.current_controlling_spiderling)
+		return
+	var/next_spiderling
+	if(spiderlings.len >= 1)
+		next_spiderling = pick(spiderlings)
+	else if(spiderlings.len <= 0)
+		next_spiderling = null
+	if(!next_spiderling)
+		spider_swarm_action.switch_to_mother()
+		return
```

Comment:
code could be better. for example you don't need next_spiderling if the len is 1 or more